### PR TITLE
admin: support HTML UI for admin post params

### PR DESCRIPTION
**Pull request by jmarantz**
Commit Message: follow-up to #19546 -- supports parameter declarations for admin POST endpoints. This required an infrastructural improvement to generate queryParams from the AdminStream, which has access to request body, as POST params from HTML form show up as POSTs params.

That in turn required that we ensure that request_headers_->getPathValue() be consistent with the path in test flows, which needed to be sorted out in the test infrastructure. Added an ASSERT to ensure they stay that way.
Additional Description:
Risk Level: low -- incremental change to support POST admin params.
Testing: //test/..., and manually ran server to verify html post behavior, fixing a couple of cases where the HTML controls didn't pass through params as expected by the handlers.
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: source/server/admin/admin.cc
Comment (jmarantz) [reply source/server/admin/admin.cc:82]: Oh wait --  what did you mean? It was using vector::insert already. I think this is would be slightly more efficient than copying the vector as is and then inserting a new beginning, as it right-size allocs it first.

Comment (pradeepcrao) [reply source/server/admin/admin.cc:82]: My hunch is insert will be faster, as it will result in moves v/s the explicit copying within prepend. Can this function not just take an r-value `strings`, insert `first` at the beginning and return it (since I don't see any other use case for it in the PR).

File: source/server/admin/admin.cc
Comment (pradeepcrao) [inline source/server/admin/admin.cc:162]: what does empty level do? I could not find this info here https://www.envoyproxy.io/docs/envoy/latest/operations/admin#post--logging

File: source/server/admin/admin.cc
Comment (jmarantz) [reply source/server/admin/admin.cc]: Sorry -- I'm confused. I *am* using *insert*.

BTW this code doesn't need to be fast at all; it just happens once on startup. I think the main argument here is not whether to use insert, but whether it's worth doing the reserve.

I don't think move vs copy makes much difference: this is a string-view.

But I see Pradeep's point: this is a local helper so I could just make it modify the strings in-place not not bother doing the copy, which makes this non-perf-critical code have fewer lines :)

File: source/server/admin/logs_handler.cc
Comment (pradeepcrao) [inline source/server/admin/logs_handler.cc:40]: Is the first parameter used for any handler, since it's now available on the AdminStream? Is this no longer needed on `MAKE_ADMIN_HANDLER`? Also several functions in admin.cc pass `path_and_query` around as arguments. Should all of these just get it from AdminStream?

File: envoy/server/admin.h
Comment (pradeepcrao) [reply envoy/server/admin.h:113]: Sorry, I meant in the help text output.

File: source/server/admin/logs_handler.cc
Comment (pradeepcrao) [reply source/server/admin/logs_handler.cc:40]: This would require AdminFilter to store the query params once parsed.

Comment (jmarantz) [reply source/server/admin/logs_handler.cc:40]: done as a pending follow-up PR: https://github.com/envoyproxy/envoy/pull/22571 -- it's not reviewable now but once this merges, that one will be non-functional so that its sprawl is easier to reason about.

**Review by pradeepcrao — DISMISSED** [review]
LGTM

**Comment by jmarantz** [comment]
Yan, can you do a senior maintainer review?

Thanks!

File: .bazelversion
Comment (yanavlasov) [inline .bazelversion:1]: This looks like merge error

**Review by yanavlasov — CHANGES_REQUESTED** [review]
/wait

File: .bazelversion
Comment (jmarantz) [reply .bazelversion]: yeah I'm not sure how that snuck in; done.

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**
